server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -220,6 +220,13 @@ final class DefaultSearchContext extends SearchContext {
             );
             queryBoost = request.indexBoost();
             this.lowLevelCancellation = lowLevelCancellation;
+            SearchSourceBuilder builder = request.source();
+            if (builder != null) {
+                Boolean requestExplain = request.source().explain();
```

Comment:
This seemed like the most logic way to get explain from the original query but would be good to validate this.